core-common/src/main/java/org/mapstruct/BeanMapping.java

```diff
@@ -38,4 +39,15 @@
      * @return the resultType to select
      */
     Class<?> resultType() default void.class;
+
+   /**
+     * A qualifier can be specified to aid the selection process of a suitable factory method. This is useful in
+     * case multiple factory method (hand written of internal) qualify and result in an 'Ambiguous factory methods'
+     * error.
+     *
+     * A qualifier is a custom annotation and can be placed on either a hand written mapper class or a method.
+     *
+     * @return the qualifiers
+     */
+    Class<? extends Annotation>[] qualifiedBy() default { };
```

Comment:
Can we add a note that either this or the other attribute is to be specified. Similar to how it's done for `@Mapping`.